File: src/weaponsprocurement/gui/WimGuiDialogDelegate.java
package weaponsprocurement.gui;

import com.fs.starfarer.api.campaign.CustomVisualDialogDelegate;
import com.fs.starfarer.api.ui.CustomPanelAPI;

final class WimGuiDialogDelegate implements CustomVisualDialogDelegate {
    private final WimGuiDialogPanel panel;

    WimGuiDialogDelegate(WimGuiDialogPanel panel) {
        this.panel = panel;
    }

    @Override
    public void init(CustomPanelAPI panel, DialogCallbacks callbacks) {
        this.panel.init(panel, callbacks);
    }

    @Override
    public WimGuiDialogPanel getCustomPanelPlugin() {
        return panel;
    }

    @Override
    public float getNoiseAlpha() {
        return 0f;
    }

    @Override
    public void advance(float amount) {
    }

    @Override
    public void reportDismissed(int option) {
        panel.reportDialogDismissed(option);
    }
}

File: src/main/kotlin/weaponsprocurement/ui/WimGuiDialogDelegate.kt
package weaponsprocurement.ui

import com.fs.starfarer.api.campaign.CustomVisualDialogDelegate
import com.fs.starfarer.api.ui.CustomPanelAPI

class WimGuiDialogDelegate(private val panel: WimGuiDialogPanel) : CustomVisualDialogDelegate {
    override fun init(panel: CustomPanelAPI, callbacks: CustomVisualDialogDelegate.DialogCallbacks) {
        this.panel.init(panel, callbacks)
    }

    override fun getCustomPanelPlugin(): WimGuiDialogPanel = panel

    override fun getNoiseAlpha(): Float = 0f

    override fun advance(amount: Float) {
    }

    override fun reportDismissed(option: Int) {
        panel.reportDialogDismissed(option)
    }
}
